### PR TITLE
DOCS-1947 New API authentication

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/api/overview.md [hunk 1]
```diff
@@ -48,33 +48,50 @@ A token is used for request authentication. The token is generated after success
 !!! info "API credentials and SSO"
     When SSO is enabled for the user, authentication for requests to Wallarm API via UUID and secret key becomes unavailable for this user. Find detailed information in the [SSO configuration](../admin-en/configuration-guides/sso/employ-user-auth.md#sso-and-api-authentication) article.
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:50]: we provide users with API call examples, please check all of them and replace auth parameters + check a couple of requests to be sure they work

File: docs/latest/api/overview.md [hunk 2]
```diff
@@ -48,33 +48,50 @@ A token is used for request authentication. The token is generated after success
...
-4. Send the required API request passing the following values:
-    * **UUID** in the `X-WallarmAPI-UUID` header parameter
-    * **Secret key** in the `X-WallarmAPI-Secret` header parameter
+## Configuring tokens
 
-![!Wallarm API credentials](../images/wallarm-api-credentials.png)
+Tokens are used for API request authentication. You can configure any number of tokens.
 
-<!-- ## API restrictions
+Normal users can:
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:65]: who are normal users?

File: docs/latest/api/overview.md [hunk 3]
```diff
@@ -48,33 +48,50 @@ A token is used for request authentication. The token is generated after success
...
+    * Set permissions, but not wider than their own.
+    * Set expiration date for the token. If set, the token will be disabled after that date.
+
+!!! info "Automatic removal"
+    All disabled tokens are automatically removed in a week after disabling.
+
+Administrators can:
+
+* See, edit, disable/enable or remove all tokens in the company account.
+* Create private or shared tokens. Shared can be used by any user in the company account.
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:81]: shared NOUN?

File: docs/latest/api/overview.md [hunk 4]
```diff
@@ -48,33 +48,50 @@ A token is used for request authentication. The token is generated after success
...
+* See or copy values of the tokens that they do not own.
+
+![!Wallarm API token](../images/api-tokens-edit.png)
+
+Consider that:
+
+* If token owner is disabled, all one's tokens are automatically disabled as well.
+* If token owner is reduced in permissions, corresponding permissions will be removed from all one's tokens.
+
+Note that some tokes are created automatically when migrating from the previous authentication approach. Permissions for the migrated legacy tokens are not displayed and cannot be changed.
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:94]: tokeNs

Comment (AnastasiaTWW) [reply docs/latest/api/overview.md:94]: "Note that some tokes are created automatically when migrating from the previous authentication approach." - what is the way for a user to understand that? the user has UUID + secret, do they stop working? how the user will identify that and what are the steps they should do to fix the problem?

File: docs/latest/api/overview.md (hunk 1 above)
Comment (MaximBashurov) [reply docs/latest/api/overview.md:50]: Fixed, but for now without testing (EKB-2141 is not ready yet).

File: docs/latest/api/overview.md [hunk 5]
```diff
@@ -78,7 +78,7 @@ Normal users can:
 Administrators can:
 
 * See, edit, disable/enable or remove all tokens in the company account.
-* Create private or shared tokens. Shared can be used by any user in the company account.
+* Create private or shared tokens. Shared tokens can be used by any user in the company account.
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:81]: add info block that admins cannot use shared tokens

File: docs/latest/api/overview.md [hunk 6]
```diff
@@ -62,7 +62,7 @@ Wallarm limits the rate of API calls to 500 requests per second. -->
 
 Tokens are used for API request authentication. You can configure any number of tokens.
 
-Normal users can:
+Non-administrator users can:
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:65]: all users + link to user docs

File: docs/latest/api/overview.md [hunk 7]
```diff
@@ -91,7 +91,7 @@ Consider that:
 * If token owner is disabled, all one's tokens are automatically disabled as well.
 * If token owner is reduced in permissions, corresponding permissions will be removed from all one's tokens.
 
-Note that some tokes are created automatically when migrating from the previous authentication approach. Permissions for the migrated legacy tokens are not displayed and cannot be changed.
+Note that some tokens are created automatically when migrating from the previous authentication approach. Permissions for the migrated legacy tokens are not displayed and cannot be changed.
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:94]: 1. explain what is previous approach
2. explain what users have to do with existing requests that use secret and UUID for auth

File: docs/latest/api/overview.md [hunk 8]
```diff
@@ -91,7 +91,7 @@ Consider that:
...
 
 ## Wallarm approach to API development and documentation
```
Comment (AnastasiaTWW) [inline docs/latest/api/overview.md:97]: + explain how SSO and new auth logic communicate

File: docs/latest/updating-migrating/older-versions/multi-tenant.md [hunk 9]
```diff
@@ -160,7 +158,7 @@ To get the list of tenants, send authenticated requests to Wallarm API. Authenti
             curl -X POST \
             https://us1.api.wallarm.com/v1/objects/client \
             -H 'content-type: application/json' \
-            -H 'x-wallarmapi-secret: YOUR_SECRET_KEY' \
+            -H "WallarmApi-Token: YOUR_TOKEN" \
             -H 'x-wallarmapi-uuid: YOUR_UUID' \
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/multi-tenant.md:162]: extra uuid

File: docs/latest/updating-migrating/older-versions/multi-tenant.md [hunk 10]
```diff
@@ -160,7 +158,7 @@ To get the list of tenants, send authenticated requests to Wallarm API. Authenti
             curl -X POST \
             https://us1.api.wallarm.com/v1/objects/client \
             -H 'content-type: application/json' \
-            -H 'x-wallarmapi-secret: YOUR_SECRET_KEY' \
+            -H "WallarmApi-Token: YOUR_TOKEN" \
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/multi-tenant.md:161]: use single quotes

File: docs/latest/updating-migrating/older-versions/multi-tenant.md [hunk 11]
```diff
@@ -169,7 +167,7 @@ To get the list of tenants, send authenticated requests to Wallarm API. Authenti
             curl -X POST \
             https://api.wallarm.com/v1/objects/client \
             -H 'content-type: application/json' \
-            -H 'x-wallarmapi-secret: YOUR_SECRET_KEY' \
+            -H "WallarmApi-Token: YOUR_TOKEN" \
             -H 'x-wallarmapi-uuid: YOUR_UUID' \
             -d '{ "filter": { "id": [<CLIENT_1_ID>, <CLIENT_2_ID>]}}'
             ```
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/multi-tenant.md:173]: change other request examples (e.g. from the section with API call examples)

Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/multi-tenant.md:173]: change the texts in the Wallarm API Swagger UI

File: docs/latest/updating-migrating/older-versions/multi-tenant.md (hunk 10 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/older-versions/multi-tenant.md]: Fixed, but noted that around the docs both single and double quotes are used in request examples.

**Pull request closed without merging**